### PR TITLE
lint: clean eslint warning

### PR DESCRIPTION
**Pull request by viet-nv**

### CODE REVIEW

File: src/hooks/useThrottle.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import _ from 'lodash'
-import { useCallback } from 'react'
+import { useEffect, useMemo, useRef } from 'react'
 
 export default function useThrottle(cb: (...args: any) => any, delay: number) {
-  return useCallback(_.throttle(cb, delay), [])
+  const cbRef = useRef(cb)
+  // use mutable ref to make useMemo not depend on `cb` dep
+  useEffect(() => {
```
Comment (hungdoansy) [inline src/hooks/useThrottle.ts:7]: Ah, we don't need useEffect here.

File: src/pages/SwapV2/index.tsx [hunk 2]
```diff
@@ -656,7 +656,7 @@ export default function Swap({ history }: RouteComponentProps) {
   }, [qs, history])
 
   // swap?inputCurrency=xxx&outputCurrency=yyy. xxx yyy not exist in chain => remove params => select default pair
-  const checkParamWrongDebounce = useCallback(debounce(checkParamWrong, 300), [])
+  const checkParamWrongDebounce = useMemo(() => debounce(checkParamWrong, 300), [checkParamWrong])
```
Comment (hungdoansy) [inline src/pages/SwapV2/index.tsx:659]: Is there any reason that this is changed from useCallback to useMemo?

File: src/state/swap/hooks.ts [hunk 3]
```diff
@@ -346,6 +346,11 @@ export const useDefaultsFromURLSearch = ():
 
   const { currencies } = useDerivedSwapInfo()
 
+  const currenciesRef = useRef(currencies)
+  useEffect(() => {
```
Comment (hungdoansy) [inline src/state/swap/hooks.ts:350]: `currenciesRef.current = currencies` is enough, no need useEffect here

Comment (viet-nv) [reply src/state/swap/hooks.ts:350]: make sense

**Comment by nguyenhoaidanh** [comment]
I am waiting for test link ...

File: src/pages/SwapV2/index.tsx (hunk 2 above)
Comment (viet-nv) [reply src/pages/SwapV2/index.tsx:659]: useCallback is designed for inline function only 

useCallback(fn, deps) is equivalent to useMemo(() => fn, deps).

Comment (hungdoansy) [reply src/pages/SwapV2/index.tsx:659]: Make sense

**Review by hungdoansy — APPROVED** [review]
LGTM

**Review by nguyenhoaidanh — APPROVED** [review]

**Pull request merged**